src/DssSpell.t.sol

```diff
@@ -5,8 +5,29 @@ pragma solidity 0.6.12;
 import "./DssSpell.t.base.sol";
 import "dss-interfaces/Interfaces.sol";
 
+interface Gem6Like {
+    function implementation() external view returns (address);
+}
+
+interface GemJoin6Like {
+    function implementations(address) external view returns (uint256);
+}
+
 contract DssSpellTest is DssSpellTestBase {
 
+    function testTUSDImplementation() public {
+        address TUSD = addr.addr("TUSD");
+        address MCD_JOIN_TUSD_A = addr.addr("MCD_JOIN_TUSD_A");
+        address implementation = Gem6Like(TUSD).implementation();
+        assertEq(GemJoin6Like(MCD_JOIN_TUSD_A).implementations(implementation), 0);
+
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done(), "DssSpellTest/spell-not-done");
+
+        assertEq(GemJoin6Like(MCD_JOIN_TUSD_A).implementations(implementation), 1);
+    }
+
```

Comment:
For extra coverage can you add a test showing that before the spell cast you can not join/exit collateral but afterwards you can?

Reply:
done